subsys/bluetooth/mesh/app_keys.c

```diff
@@ -106,8 +106,8 @@ static void store_app_key(uint16_t app_idx)
 	key.net_idx = app->net_idx,
 	key.updated = app->updated,
 
-	memcpy(key.val[0], app->keys[0].val, 16);
-	memcpy(key.val[1], app->keys[1].val, 16);
+	memcpy(KEY_REF(key.val[0]), KEY_REF(app->keys[0].val), sizeof(bt_mesh_key_t));
```

Comment:
The packed data are used in some places with `memcpy` (for example in cdb). `memcpy` is designed to recognize packed attribute during compile time and build-in correct code. If separate function is implemented then unaligned access hardfault will be possible since function body will lose attribute.
Using memcpy as hardfault safe here.

Reply:
Exactly a simple assignment will generate unaligned access hardfault. Since type is more than 1 byte the compiler will use multibyte instructions to optimize access that might cause assertion (depends how local variable will be located in the stack). Btw. the compiler understands the potential problem with it and generates warnings in all such assignments (see my latest changes for app and net key restoring from cdb). I had to add one more extra copying.